chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -489,6 +531,29 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         Ok(())
     }
 
+    fn register_tokens(&mut self, tx: &Transaction) -> Result<(), TokensError> {
+        let token_id = token_id(tx).ok_or(TokensError::TokenIdCantBeCalculated)?;
+
+        let _tokens_op = match self.tokens_cache.entry(token_id) {
+            Entry::Occupied(entry) => entry.into_mut(), // TODO: Tokens already issued? Must we return error?
```

Comment:
This should return an error with an invariant being broken as this should never happen.